storage/parquet/datasource.go

```diff
@@ -417,39 +435,151 @@ func (rs *RecordStream) RunWorkerInternal(ctx context.Context, tx storage.StateT
 
 	batch := make([]*execution.Record, 0)
 	for i := 0; i < rs.batchSize; i++ {
-		record, err := rs.readRecordFromFile()
+		recordValues, err := rs.readRecordFromFile()
 		if err == execution.ErrEndOfStream {
 			break
 		} else if err != nil {
 			return errors.Wrap(err, "couldn't read record from parquet file")
 		}
 
-		aliasedRecord := make(map[octosql.VariableName]octosql.Value)
-		for k, v := range record {
-			if str, ok := v.(string); ok {
-				parsed, err := time.Parse(time.RFC3339, str)
-				if err == nil {
-					v = parsed
-				}
-			}
+		octoValues := make([]octosql.Value, len(rs.columns))
+		for i, v := range recordValues {
 			if int96, ok := v.(parquet.Int96); ok {
 				v = int(binary.LittleEndian.Uint64(int96[:8]))
 			}
-			aliasedRecord[octosql.NewVariableName(fmt.Sprintf("%s.%s", rs.alias, k))] = octosql.NormalizeType(v)
-		}
+			// Info here: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
+			if rs.columnsLogicalTypes[i] != nil && v != nil {
+				var convert func(interface{}) interface{}
+
+				logicalType := rs.columnsLogicalTypes[i]
+				switch {
```

Comment:
Why is this an empty `switch` after a declaration and not `switch logicalType := rs.columnsLogicalTypes[i]`?

Reply:
Because the syntax you proposed only works for type switches.